java/arcs/core/storage/ProxyInterface.kt

```diff
@@ -108,16 +108,15 @@ interface StorageCommunicationEndpoint<Data : CrdtData, Op : CrdtOperation, Cons
     fun close()
 }
 
-/** Provider of a [StorageCommunicationEndpoint]. */
-interface StorageCommunicationEndpointProvider<Data : CrdtData, Op : CrdtOperation, ConsumerData> {
+/** Provider of a [StorageCommunicationEndpoint]s. */
+interface StorageEndpointProvider {
     /**
-     * Implementers should return a [StorageCommunicationEndpoint] that signals information back to
-     * the agent using the provided [callback].
+     * Returns a communications channel to an [ActiveStore] that reflects the provided
+     * [StoreOptions]. This is not necessarily an [ActiveStore] implementation, though a basic
+     * implementation may provide a simple wrapper around an in-process instance of [ActiveStore].
      */
-    fun getStorageEndpoint(
+    fun <Data : CrdtData, Op : CrdtOperationAtTime, ConsumerData> getStorageEndpoint(
+        storeOptions: StoreOptions,
         callback: ProxyCallback<Data, Op, ConsumerData>
-    ): StorageCommunicationEndpoint<Data, Op, ConsumerData>
-
-    /** Return the [StorageKey] that the store behind this implementation is representing. */
-    val storageKey: StorageKey
+    ): StorageEndpoint<Data, Op, ConsumerData>
```

Comment:
This change is the centerpiece of this PR.

Reply:
If it always creates a new storage endpoint, it should be called `createStorageEndpoint`.. if it re-uses them (or is allowed to re-use them, contractually by the interface) - `getStorageEndpoint` is fine.

Reply:
Good call, should be create.